### DISCUSSION TITLE
NSFW Content

### DISCUSSION THREAD

**Discussion by ghost** [mastodon/mastodon#20796] [General]
Hallo everyone.

I see a big content problem, sitting here in germany.
I can visit any Mastodon instance i want and search for tags (url: /tags/nsfw) or take a look into a persons profile, search for people responding to their posts and maybe find nsfw content in their posts.
The problem is, that this content is delivered from the cache of that instance server.

I'am not a lawyer, but i think in germany this is illegal, because you are hosting nsfw stuff and maybe p*rn or even harder, more illegal material. It is stored on that instance server.

So, how can we fix that issue? Can a functionality to Mastodon be added, which hides everything on that instance to guests, except the server informations like "about" and "dataprotection"?

**Comment by trwnh** [comment]
there is DISALLOW_UNAUTHENTICATED_API_ACCESS in 4.0

**Reply by ghost** [reply]
Oh, thats a great thing. Is there a documentation how to activate that and where the disadvantages are?

**Reply by ghost** [reply]
ok, i enabled it in .env.production. This flag also disabled the contact details, like, the complete api for not logged in users.
This is no solution because contact details must be present at any time. Also the ui is flooded with 401 error messages.
